spec/models/container_label_tag_mapping_spec.rb

```diff
@@ -1,123 +1,188 @@
-  before :all do
-    ContainerLabelTagMapping.drop_cache
+
+  # All-in-one
+  def map_to_tags(model_name, labels_kv)
+    to_tags(map_labels(model_name, labels_kv))
   end
 
   context "with empty mapping" do
     it "does nothing" do
-      expect(ContainerLabelTagMapping.tags_for_entity(node)).to be_empty
-      expect(ContainerLabelTagMapping.mappable_tags).to be_empty
+      expect(ContainerLabelTagMapping.cache).to be_empty
+      expect(map_labels('ContainerNode',
+                        'foo'  => 'bar',
+                        'quux' => 'whatever')).to be_empty
     end
   end
 
   context "with 2 mappings for same label" do
-    before do
+    before(:each) do
       FactoryGirl.create(:container_label_tag_mapping, :only_nodes, :label_value => 'value-1', :tag => tag1)
       FactoryGirl.create(:container_label_tag_mapping, :only_nodes, :label_value => 'value-1', :tag => tag2)
     end
 
-    it "tags_for_entity returns 2 tags" do
-      label(node, 'name', 'value-1')
-      expect(ContainerLabelTagMapping.tags_for_entity(node)).to contain_exactly(tag1, tag2)
-    end
-
-    it "tags_for_label returns same tags" do
-      label_obj = OpenStruct.new(:resource_type => 'ContainerNode',
-                                 :name          => 'name',
-                                 :value         => 'value-1')
-      expect(ContainerLabelTagMapping.tags_for_label(label_obj)).to contain_exactly(tag1, tag2)
+    it "map_labels returns 2 tags" do
+      expect(map_labels('ContainerNode', 'name' => 'value-1')).to contain_exactly(
+        {:tag_id => tag1.id},
+        {:tag_id => tag2.id}
+      )
+      expect(map_to_tags('ContainerNode', 'name' => 'value-1')).to contain_exactly(tag1, tag2)
     end
   end
 
   context "with any-value and specific-value mappings" do
-    before do
+    before(:each) do
       FactoryGirl.create(:container_label_tag_mapping, :tag => cat_tag)
       FactoryGirl.create(:container_label_tag_mapping, :label_value => 'value-1', :tag => tag1)
-      FactoryGirl.create(:container_label_tag_mapping, :label_value => 'value-1', :tag => tag_under_cat)
-      # Force a tag to exist that we don't map to (for testing .mappable_tags).
-      tag2
+      FactoryGirl.create(:container_label_tag_mapping, :label_value => 'value-1', :tag => tag2)
+      # Force a tag to exist that we don't map to (for testing .controls_tag?).
+      tag3
     end
 
     it "prefers specific-value" do
-      label(node, 'name', 'value-1')
-      expect(ContainerLabelTagMapping.tags_for_entity(node)).to contain_exactly(tag1, tag_under_cat)
+      expect(map_to_tags('ContainerNode', 'name' => 'value-1')).to contain_exactly(tag1, tag2)
     end
 
     it "creates tag for new value" do
-      expect(ContainerLabelTagMapping.mappable_tags).to contain_exactly(tag1, tag_under_cat)
+      expect(ContainerLabelTagMapping.controls_tag?(tag1)).to be true
+      expect(ContainerLabelTagMapping.controls_tag?(tag2)).to be true
```

Comment:
Yes, thanks.  Had to create new tag since this whole category is target of any-value mapping.